### PR TITLE
Adding Pdf Worker

### PR DESCRIPTION
**Pull request by Bearloggs**
Pdf worker permits to analyze PDF files for potential malicious content and suspicious elements.

Features:
* **Encoding Check**
Checks if the PDF is encrypted
* **JavaScript Extraction**
Extracts JavaScript code embedded within the PDF, including /JS and /JavaScript entries. Truncates long scripts for better readability
* **Suspicious Object Detection**
Identifies suspicious objects such as /AA (Additional Actions) and /OpenAction. Truncates long actions for better readability
* **Embedded File Detection**
Detects and reports embedded files within the PDF, providing the embedded file metadata.

### CODE REVIEW

File: pandora/workers/pdf.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+    check_encoding: bool
+    check_javascript: bool
+    check_suspicious_object: bool
+    check_embedded_files: bool
+
+    def _is_encoding(self, file_path: str) -> bool:
+        try:
+            doc = fitz.open(file_path)
+            return doc.is_encrypted
```
Comment (Rafiot) [inline pandora/workers/pdf.py:23]: Should this method be called `_is_encrypted` instead? Any reason you call the method `_is_encoded`?

File: pandora/workers/pdf.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+            doc = fitz.open(file_path)
+            for item in range(doc.embfile_count()):
+                embedded_files.append(doc.embfile_info(item))
+
+        except Exception as e:
+            self.logger.warning(f'Unable to detect embedded files in PDF file: {e}')
+
+        return embedded_files
+
+    def _truncate_string(self, string: str, max_length: int = 100) -> str:
```
Comment (Rafiot) [inline pandora/workers/pdf.py:90]: I'm not sure we should truncate the strings in the module. It should probably be done in the UI directly (so the user has a chance to get to the complete content)

File: pandora/workers/pdf.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+
+    def analyse(self, task: Task, report: Report, manual_trigger: bool=False) -> None:
+        if not task.file.is_pdf:
+            report.status = Status.NOTAPPLICABLE
+            return
+
+        self.logger.debug(f'Analysing PDF file {task.file.path}...')
+        try:
+            is_encoded = ""
```
Comment (Rafiot) [inline pandora/workers/pdf.py:104]: Should be `is_encrypted`, right?

**Review by Rafiot — CHANGES_REQUESTED** [review]
Thank you for the PR, it looks great.

Just a few notes before I merge it. And if you want to fix the typing issues (just run `mypy .`), go for it, but otherwise I'll take care of it.

File: pandora/workers/pdf.py (hunk 3 above)
Comment (Bearloggs) [reply pandora/workers/pdf.py]: You're right. I change the variable name.

File: pandora/workers/pdf.py (hunk 2 above)
Comment (Bearloggs) [reply pandora/workers/pdf.py]: You're right, it is better to handle through the UI. But do you know how to do it? Thanks

**Comment by Bearloggs** [comment]
> Thank you for the PR, it looks great.
> 
> Just a few notes before I merge it. And if you want to fix the typing issues (just run `mypy .`), go for it, but otherwise I'll take care of it.

I fix the typing issue with mypy also.

File: pandora/workers/pdf.py (hunk 1 above)
Comment (Rafiot) [reply pandora/workers/pdf.py:23]: Thanks!

**Comment by Rafiot** [comment]
Merged, thank you!

**Pull request merged**